File: 1-easy/problems/1805-number-of-different-integers-in-a-string/solution.py
class Solution:
    def numDifferentIntegers(self, word):
        """
        :type word: str
        :rtype: int
        """
        result, num = set(), None
        for i in range(len(word)+1):
            c = word[i] if i < len(word) else ' '
            if c.isdigit():
                num = 10*num+int(c) if num is not None else int(c)
            elif num is not None:
                result.add(num)
                num = None
        return len(result)

File: 1-easy/problems/1805-number-of-different-integers-in-a-string/solution.ts
function numDifferentIntegers(word: any): boolean | number | string | any {
  """
  :type word: str
  :rtype: int
  """
  result, num = set(), null
  for (i in Array.from({length: (word}, (_, i) => i).length+1)) {
      c = word[i] if i < (word).length else ' '
      if (c.isdigit()) {
          num = 10*num+int(c) if num is not null else int(c)
      } else if (num is not null:
          result.add(num)
          num = null
  return (result).length;
}

export { numDifferentIntegers };
